src/dbc_content.rs

```diff
@@ -39,33 +46,94 @@ impl BlindedOwner {
     }
 }
 
+/// Contains amount and Pedersen Commitment blinding factor which
+/// must be kept secret (encrypted) in the DBC.
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct AmountSecrets {
+    pub amount: u64,
+    pub blinding_factor: Scalar,
+}
+
+impl AmountSecrets {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut v: Vec<u8> = Default::default();
+        v.extend(&self.amount.to_le_bytes());
+        v.extend(&self.blinding_factor.to_bytes());
+        v
+    }
+
+    #[allow(clippy::transmute_ptr_to_ref)]
+    fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
```

Comment:
yeah, I considered that, though when you have a Vec<u8> you are passing in, it just gives the caller a conversion headache and they still have to deal with a possible error.  I have learned how to do it without the macro and clippy warning though.  eg:   
```
let mut b = [0u8; 8];
b.copy_from_slice(&bytes[0..8]);
```

anyway, I will revisit it.

Reply:
ok, I changed it.  But then I added a ::from_bytes_ref() also because like I said the caller has a Vec<u8> that it gets as result from PublicKeySet::decrypt().